telegram/request/_baserequest.py

```diff
@@ -312,20 +312,24 @@ async def _request_wrapper(
 
             message += f"\nThe server response contained unknown parameters: {parameters}"
 
-        if code == HTTPStatus.FORBIDDEN:
+        if code == HTTPStatus.FORBIDDEN:  # 403
             raise Forbidden(message)
-        if code in (HTTPStatus.NOT_FOUND, HTTPStatus.UNAUTHORIZED):
+        if code in (HTTPStatus.NOT_FOUND, HTTPStatus.UNAUTHORIZED):  # 404 and 401
             # TG returns 404 Not found for
             #   1) malformed tokens
             #   2) correct tokens but non-existing method, e.g. api.tg.org/botTOKEN/unkonwnMethod
             # We can basically rule out 2) since we don't let users make requests manually
             # TG returns 401 Unauthorized for correctly formatted tokens that are not valid
-            raise InvalidToken(message)
-        if code == HTTPStatus.BAD_REQUEST:
+
+            # retrieve token from the url-
+            start = url[url.find("/bot") + 4 :]
+            token = start[: start.find("/")]
+            raise InvalidToken(message + f"\nThe token: {token} was rejected by the server.")
```

Comment:
doing this at the request level seems rather unclean to me, tbh, especially since we have to extract the token from the url again. what would you rather think about doing something like

```
try:
    ...
except InvalidToken as exc:
    raise InvalidToken(f"The token {token} was …") from exc
```

in `Bot._post` - or even just once in `Bot.initialize`? I guess the latter should already suffice …